ReactNativeClient/lib/services/KeymapService.ts

```diff
@@ -0,0 +1,246 @@
+		{ accelerator: 'Shift+Cmd+N', command: 'focusElementNoteTitle' },
+		{ accelerator: 'Shift+Cmd+B', command: 'focusElementNoteBody' },
+		{ accelerator: 'Option+Cmd+S', command: 'toggleSidebar' },
+		{ accelerator: 'Cmd+L', command: 'toggleVisiblePanes' },
+		{ accelerator: 'Cmd+0', command: 'zoomActualSize' },
+		{ accelerator: 'Cmd+E', command: 'startExternalEditing' },
+		{ accelerator: 'Option+Cmd+T', command: 'setTags' },
+		{ accelerator: 'Cmd+G', command: 'gotoAnything' },
+		{ accelerator: 'F1', command: 'help' },
+	],
+	default: [
+		{ accelerator: 'Ctrl+N', command: 'newNote' },
+		{ accelerator: 'Ctrl+T', command: 'newTodo' },
+		{ accelerator: 'Ctrl+S', command: 'synchronize' },
+		{ accelerator: 'Ctrl+P', command: 'print' },
+		{ accelerator: 'Ctrl+Q', command: 'quit' },
+		{ accelerator: 'Ctrl+Alt+I', command: 'insertTemplate' },
+		{ accelerator: 'Ctrl+C', command: 'textCopy' },
+		{ accelerator: 'Ctrl+X', command: 'textCut' },
+		{ accelerator: 'Ctrl+V', command: 'textPaste' },
+		{ accelerator: 'Ctrl+A', command: 'textSelectAll' },
+		{ accelerator: 'Ctrl+B', command: 'textBold' },
+		{ accelerator: 'Ctrl+I', command: 'textItalic' },
+		{ accelerator: 'Ctrl+K', command: 'textLink' },
+		{ accelerator: 'Ctrl+`', command: 'textCode' },
+		{ accelerator: 'Ctrl+Shift+T', command: 'insertDateTime' },
+		{ accelerator: 'F6', command: 'focusSearch' },
+		{ accelerator: 'Ctrl+F', command: 'showLocalSearch' },
+		{ accelerator: 'Ctrl+Shift+S', command: 'focusElementSideBar' },
+		{ accelerator: 'Ctrl+Shift+L', command: 'focusElementNoteList' },
+		{ accelerator: 'Ctrl+Shift+N', command: 'focusElementNoteTitle' },
+		{ accelerator: 'Ctrl+Shift+B', command: 'focusElementNoteBody' },
+		{ accelerator: 'F10', command: 'toggleSidebar' },
+		{ accelerator: 'Ctrl+L', command: 'toggleVisiblePanes' },
+		{ accelerator: 'Ctrl+0', command: 'zoomActualSize' },
+		{ accelerator: 'Ctrl+E', command: 'startExternalEditing' },
+		{ accelerator: 'Ctrl+Alt+T', command: 'setTags' },
+		{ accelerator: 'Ctrl+,', command: 'config' },
+		{ accelerator: 'Ctrl+G', command: 'gotoAnything' },
+		{ accelerator: 'F1', command: 'help' },
+	],
+};
+
+interface KeymapItem {
+	accelerator: string;
+	command: string;
+}
+
+interface Keymap {
+	[command: string]: KeymapItem;
+}
+
+class KeymapService extends BaseService {
+	private keymap: Keymap;
+	private defaultKeymap: KeymapItem[];
+
+	constructor() {
+		super();
+
+		// Automatically initialized for the current platform
+		this.initialize();
+	}
+
+	initialize(platform: string = shim.platformName()) {
+		this.keysRegExp = keysRegExp;
+		switch (platform) {
+		case 'darwin':
+			this.defaultKeymap = defaultKeymap.darwin;
+			this.modifiersRegExp = modifiersRegExp.darwin;
+			break;
+		case 'win32':
+		case 'linux':
+			this.defaultKeymap = defaultKeymap.default;
+			this.modifiersRegExp = modifiersRegExp.default;
+			break;
+		default:
+			throw new Error(`Unsupported platform: ${platform}`);
```

Comment:
Why throw an exception here? It should be either "darwin" or "default". There's no need to list all the supported platforms. That way if it's a platform we don't know about it will work instead of crashing for no particular reason.